src/extensions/arithmetic/int_ops.rs

```diff
@@ -0,0 +1,517 @@
+//! Basic integer operations.
+
+use itertools::Itertools;
+use smol_str::SmolStr;
+
+use super::super::logic::bool_type;
+use super::int_types::{get_width, int_type, INT_PARAM};
+use crate::{
+    resource::{ResourceSet, SignatureError},
+    types::{type_param::TypeArg, HashableType, SimpleRow, SimpleType, TypeRow},
+    Resource,
+};
+
+/// The resource identifier.
+pub const fn resource_id() -> SmolStr {
+    SmolStr::new_inline("arithmetic.int")
+}
+
+fn iwiden_sig(
+    arg_values: &[TypeArg],
+) -> Result<(SimpleRow, SimpleRow, ResourceSet), SignatureError> {
+    let (arg0, arg1) = arg_values.iter().collect_tuple().unwrap();
```

Comment:
a function like

```rust

fn arg_arr<const N: usize>(args: &[TypeArg]) -> [&TypeArg; N] {
    args.iter().collect_vec().try_into().unwrap()
}
...
let [arg0, arg1] = arg_arr(arg_values);
```
may be of use

Reply:
Yes, though not specific to this module; what would be the best place for such a "utility" function?

Reply:
could even be this I guess
```rust

fn collect_array<const N: usize, T>(args: &[T]) -> [&T; N] {
    args.iter().collect_vec().try_into().unwrap()
}

```